frappe/database/mariadb/database.py

```diff
@@ -127,7 +125,15 @@ def get_connection_settings(self) -> dict:
 		if self.cur_db_name:
 			conn_settings["database"] = self.cur_db_name
 
-		if self.port:
+		if self.socket:
+			conn_settings["unix_socket"] = self.socket
+		else:
+			conn_settings["host"] = self.host
+
+		if self.password:
+			conn_settings["password"] = self.password
+
+		if not self.socket and self.port:
```

Comment:
Can move this to a nested if on line 131

Reply:
Done in https://github.com/frappe/frappe/pull/22548/commits/d14d3fa28485adf8fac0e44c723721052dabab43